### DISCUSSION TITLE
Protection for pushing event to plausible

### DISCUSSION THREAD

**Discussion by archerbj** [plausible/analytics#1146] [Anything Else]
Hello, We know  API Key is required for querying data from plausible. But from my view, there are no requirements for pushing the event to plausible. I have tried with the postman, just put the data in the post body. And plausible will accept it. 

Am I missing something about this? or this is worked as design?

IMHO, malicious attack could mess up our actual data by using this

**Comment by ukutaht** [comment]
> IMHO, malicious attack could mess up our actual data by using this

Absolutely true and it's also true for every other web analytics tool that works in the browser. You cannot hide anything in the browser. The only way you could secure the endpoint is if you sent them through your own backend and add the API key there where it can be kept secret. This is much much more work than just installing the snippet in your browser.

We could add some kind of endpoint protection ioption n the future if you want to go that route and manage your own server that proxies the requests with an added API key.

Is this something you would implement on your end to protect your dashboard from having misleading stats?
